FirebaseStorageSwift/Tests/Integration/StorageIntegration.swift

```diff
@@ -0,0 +1,620 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseCore
+import FirebaseStorage
+import FirebaseStorageSwift
+import XCTest
+
+class StorageIntegration: XCTestCase {
+  var app: FirebaseApp!
+  var storage: Storage!
+  static var once = false
+
+  override class func setUp() {
+    FirebaseApp.configure()
+  }
+
+  override func setUp() {
+    super.setUp()
+    app = FirebaseApp.app()
+    storage = Storage.storage(app: app!)
+
+    if !StorageIntegration.once {
+      StorageIntegration.once = true
+      let setupExpectation = expectation(description: "setUp")
+
+      let largeFiles = ["ios/public/1mb"]
+      let emptyFiles =
+        ["ios/public/empty", "ios/public/list/a", "ios/public/list/b", "ios/public/list/prefix/c"]
+      setupExpectation.expectedFulfillmentCount = largeFiles.count + emptyFiles.count
+
+      do {
+        let bundle = Bundle(for: StorageIntegration.self)
+        let filePath = try XCTUnwrap(bundle.path(forResource: "1mb", ofType: "dat"),
+                                     "Failed to get filePath")
+        let data = try XCTUnwrap(try Data(contentsOf: URL(fileURLWithPath: filePath)),
+                                 "Failed to load file")
+
+        for largeFile in largeFiles {
+          let ref = storage.reference().child(largeFile)
+          _ = ref.putData(data) { result in
+            switch result {
+            case let .success(metadata):
+              XCTAssertEqual(metadata.name, "1mb")
+            case let .failure(error):
+              XCTFail("Unexpected error \(error) from putData")
+            }
+            setupExpectation.fulfill()
+          }
+        }
+        for emptyFile in emptyFiles {
+          let ref = storage.reference().child(emptyFile)
+          _ = ref.putData(data) { result in
+            switch result {
```

Comment:
You could cut down on the verbosity of these tests by creating assertion helpers. Something like

```swift
func assertResultNotNil<T>(
    _ result: Result<T, Error>,
     file: StaticString = #file, line: UInt = #line) {
  switch result {
    case let .success(value):
      XCTAssertNotNil(value, file: file, line: line);
    case let .failure(error):
      XCTFail("Unexpected error \(error)")
  }
}
```

Now this whole mess can just be:

```swift
_ = ref.putData(data) { result in
  assertResultNotNil(result);
  setupExpectation.fulfill();
}
```

There are a few variations in here (equality, failure happened) but it seems like this would make these tests much more readable.

Reply:
Added assertResultSuccess to get the most common case.